kairon/shared/channels/mail/processor.py

```
@@ -0,0 +1,240 @@
+                messages = json.dumps(messages)
+                info = await self.llm_processor.predict(messages,
+                                                        self.bot_settings.user,
+                                                        system_prompt=system_prompt,
+                                                        context_prompt=context_prompt,
+                                                        similarity_prompt=[],
+                                                        hyperparameters=self.hyperparameters)
+                classifications = MailProcessor.extract_jsons_from_text(info["content"])[0]
+                return classifications
+            except Exception as e:
+                logger.error(str(e))
+                raise AppException(str(e))
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)